enchant: allow toggling individual dictionaries

WhyNotHugo opened the pull request:
Allow disabling and enabling individual dictionaries with the Enchant/Hunspell backend.

This is mostly relevant in shared systems (where different users want to enable different sets of languages) or on distributions which install an exorbitant amount of dictionaries by default.

WhyNotHugo committed:
enchant: allow toggling individual dictionaries

Allow disabling and enabling individual dictionaries with the
Enchant/Hunspell backend.

This is mostly relevant in shared systems (where different users want to
enable different sets of languages) or on distributions which install an
exorbitant amount of dictionaries by default.

The SupportsToggleDictionaries() method should be used to determine
whether this is supported in the current environment. The macOS and
Windows both currently return false, with the latter apparently being
feasible but out of scope.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/desktop-app/lib_spellcheck?pullRequest=32) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/desktop-app/lib_spellcheck?pullRequest=32) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/desktop-app/lib_spellcheck?pullRequest=32) it.</sub>

ilya-fedin commented:
Do we still need those workarounds or they could be reverted now?
https://github.com/desktop-app/lib_spellcheck/commit/dd6719be0182465e6d6c6f91af9a9be44d60fd61
https://github.com/desktop-app/lib_spellcheck/commit/0f030ddb2e20cbfb6c020c59f2e35b556c92854b
https://github.com/desktop-app/lib_spellcheck/commit/cc5eaf385658312baa9fa9fd504ece2e6883900f

WhyNotHugo committed:
enchant: remove obsolete workarounds

These workarounds were only necessary when choosing dictionaries was not
possible. See: https://github.com/desktop-app/lib_spellcheck/pull/32
